### PR TITLE
CI: CircleCI 2.0 with single custom (primary) docker container

### PR DESCRIPTION
**Pull request by klauer**
Reference:
https://github.com/klauer/caproto-docker
https://hub.docker.com/r/klauer/caproto-docker/

Docker image is still messy because it was made lazily, but it works!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            cd
+            wget https://raw.githubusercontent.com/pyepics/testioc/master/simulator.py
+            python simulator.py
+          background: true
+
+      - run:
+          name: run caproto asyncio repeater in background
+          command: |
+            . caproto-venv/bin/activate
+            python scripts/caproto-repeater
```
Comment (danielballan) [inline .circleci/config.yml]: This works, but also `caproto-repeater` should already be on the PATH, put there by setup.py.

**Comment by danielballan** [comment]
This looks really nice. Can we leave the Travis config and that handy install_epics... script intact though?

**Comment by klauer** [comment]
Yes, they can be left intact. It was just getting in the way when I pushed the CircleCI stuff 10x in a few minutes.

**Comment by klauer** [comment]
@danielballan any input on the post-success steps and how they should fit into CircleCI 2.0? For reference, https://circleci.com/docs/2.0/configuration-reference/

Edit: I guess the only real downside here is that we don't automatically have a matrix of Python versions to test against... hmm...

**Pull request closed without merging**